oak-run-commons/src/test/java/org/apache/jackrabbit/oak/fixture/DataStoreUtilsTest.java

```diff
@@ -0,0 +1,259 @@
+        container = azuriteDockerRule.getContainer(CONTAINER_NAME);
+    }
+
+    @After
+    public void cleanup() throws StorageException {
+        if (container != null) {
+            container.deleteIfExists();
+        }
+    }
+
+    @Test
+    public void delete_non_existing_container_azure_connection_string() throws Exception {
+        container.deleteIfExists();
+        ListAppender<ILoggingEvent> logAppender = subscribeAppender();
+        final String azureConnectionString = String.format(AZURE_CONNECTION_STRING, AzuriteDockerRule.ACCOUNT_NAME, AzuriteDockerRule.ACCOUNT_KEY, azuriteDockerRule.getBlobEndpoint());
+
+        DataStoreUtils.deleteAzureContainer(getConfigMap(azureConnectionString, null, null, null, null, null, null, null), CONTAINER_NAME);
+        validate(Arrays.asList(AUTHENTICATE_VIA_AZURE_CONNECTION_STRING_LOG,
+                        REFRESH_TOKEN_EXECUTOR_SHUTDOWN_LOG, String.format(CONTAINER_DOES_NOT_EXIST_MESSAGE, CONTAINER_NAME)),
+                Arrays.asList(AUTHENTICATE_VIA_SERVICE_PRINCIPALS_LOG, AUTHENTICATE_VIA_SAS_TOKEN_LOG, AUTHENTICATE_VIA_ACCESS_KEY_LOG),
+                getLogMessages(logAppender));
+        unsubscribe(logAppender);
+    }
+
+    @Test
+    public void delete_existing_container_azure_connection_string() throws Exception {
+        final String azureConnectionString = String.format(AZURE_CONNECTION_STRING, AzuriteDockerRule.ACCOUNT_NAME, AzuriteDockerRule.ACCOUNT_KEY, azuriteDockerRule.getBlobEndpoint());
+        ListAppender<ILoggingEvent> logAppender = subscribeAppender();
+
+        DataStoreUtils.deleteAzureContainer(getConfigMap(azureConnectionString, null, null, null, null, null, null, null), CONTAINER_NAME);
+        validate(Arrays.asList(AUTHENTICATE_VIA_AZURE_CONNECTION_STRING_LOG, REFRESH_TOKEN_EXECUTOR_SHUTDOWN_LOG,
+                        String.format(CONTAINER_DELETED_MESSAGE, CONTAINER_NAME)),
+                Arrays.asList(AUTHENTICATE_VIA_SERVICE_PRINCIPALS_LOG, AUTHENTICATE_VIA_SAS_TOKEN_LOG, AUTHENTICATE_VIA_ACCESS_KEY_LOG),
+                getLogMessages(logAppender));
+        unsubscribe(logAppender);
+    }
+
+    @Test
+    public void delete_non_existing_container_service_principal() throws Exception {
+        final String accountName = getEnvironmentVariable(AZURE_ACCOUNT_NAME);
+        final String clientId = getEnvironmentVariable(AZURE_CLIENT_ID);
+        final String clientSecret = getEnvironmentVariable(AZURE_CLIENT_SECRET);
+        final String tenantId = getEnvironmentVariable(AZURE_TENANT_ID);
+
+        Assume.assumeNotNull(accountName);
+        Assume.assumeNotNull(clientId);
+        Assume.assumeNotNull(clientSecret);
+        Assume.assumeNotNull(tenantId);
+
+        ListAppender<ILoggingEvent> logAppender = subscribeAppender();
+
+        DataStoreUtils.deleteAzureContainer(getConfigMap(null, accountName, null, null, null, clientId, clientSecret, tenantId), CONTAINER_NAME);
```

Comment:
how are we making sure here that the CONTAINER_NAME is non existing ? If we are not, how is the test passing ?

Reply:
using a random container name now